internal/log_analysis/rules_engine/src/enriched_event.py

```
@@ -0,0 +1,73 @@
+
+    def __init__(self, event: Mapping, data_model: DataModel):
+        """Create data model lookups
+
+        Args:
+            event: Dictionary representing the event
+            data_model: the data model used for the LogType associated with this event
+        """
+        self.logger = get_logger()
+        self._data = event
+        self.data_model = data_model
+
+    def __getitem__(self, key: str) -> Any:
+        return self._data[key]
+
+    def __iter__(self) -> Iterator:
+        return iter(self._data)
+
+    def __len__(self) -> int:
+        return len(self._data)
+
+    def udm(self, key: str) -> Any:
+        """Converts standard data model field to logtype field"""
+        # access values via normal logType fields
+        if key in self._data.keys():
+            return self.get(key)
+        # access values via standardized fields
+        if key in self.data_model.fields.keys():
+            # we are dealing with a jsonpath
+            json_path = self.data_model.fields.get(key)
+            if json_path:
+                matches = json_path.find(self._data)
+                if len(matches) == 1:
+                    return matches[0].value
+                if len(matches) > 1:
+                    # if we wanted, we could also extract the field names that match
+                    self.logger.error('JSONPath [%s] in DataModel [%s] matched multiple fields.', json_path, self.data_model.data_model_id)
+                    raise Exception(
+                        'JSONPath [{}] in DataModel [{}], matched multiple fields.'.format(json_path, self.data_model.data_model_id)
+                    )
+        if key in self.data_model.methods.keys():
+            # we are dealing with method
+            method = self.data_model.methods.get(key)
+            if callable(method):
+                # return self.get(method(self._data))
+                return method(self._data)
+        # no matches, return None by default
+        return None
```

I wasn't sure if this should raise an error or just return `None` - I went with `None` to prevent errors here.  I figured if a user tries to use the value returned and it is `None` there will be an error in the rule logic anyway.